src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -279,9 +283,9 @@ internal static DeclarationModifiers AdjustModifiersForAnInterfaceMember(Declara
                 }
             }
 
-            if ((mods & DeclarationModifiers.AccessibilityMask) == 0)
+            if (noExplicitAccessibilityModifiers)
             {
-                if ((mods & DeclarationModifiers.Partial) == 0 && !isExplicitInterfaceImplementation)
+                if (!plainPartialMethod && !isExplicitInterfaceImplementation)
```

Comment:
nit: I was wondering if doing this part first, could simplify logic at all, since we would be able to just say: is it private? then do not make it virtual.